src/utils/components/withMatchMedia.jsx

```diff
@@ -0,0 +1,136 @@
+import React from 'react';
+import { MEDIA_QUERIES } from '../designConstants';
+
+/**
+ * @param {String} breakpoint - name of breakpoint (for example, 'medium')
+ * @returns {String} - state property name (for example, isAtMediumUp)
+ */
+export const getStateNameByBreakpoint = breakpoint => {
+	const capitalizedBp = breakpoint[0].toUpperCase() + breakpoint.slice(1);
```

Comment:
prefer a string template instead of `+` string concatenation, and prefer string methods instead of array methods to manipulate strings (just because they're a bit more explicit about the data type)

```js
return `isAt${breakpoint.substr(1, 0).toUpperCase()}{breakpoint.substr(1)}`
// or if you rename the arg to be shorter
return `isAt${bp.substr(1, 0).toUpperCase()}{bp.substr(1)}`;
```